submit/samples/sample_checklist/sample_checklist_errors.md

```diff
@@ -0,0 +1,24 @@
+
+| **XML validation (old)**                                                                                                                                                         | **BioSamples validation (New)**                                                                                                                                                            |
+|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Error at {<sample-name>}. Invalid units "{value-of-units}" for field "{<attribute-name>}". Allowed units are "{<list-of-allowed-values>}"                                        | sample:{<sample-name>}; attribute:{<path-to-attribute>}; reason:must be equal to one of the allowed values:{list-of-allowed-values}                                                        |
+| Missing units for field "{<attribute-name>}". Allowed units are "{<list-of-allowed-values>}"                                                                                     | sample:{<sample-name>}; attribute:{<path-to-attribute>}; reason:must have required property 'unit'                                                                                         |
+| Error at {<sample-name>}. The length of field "{<attribute-name>}" value "{<value-provided>}" exceeds the maximum allowed length "{<maximum-length-of-field>}".                  | sample:{<sample-name>}; attribute:{<path-to-attribute>}; reason:must NOT have more than {<maximum-length-of-field>} characters                                                             |
+| Error at {<sample-name>}. The value "{<value-provided>}" for field "{<attribute-name>}" does not match regular expression "{<pattern>}"                                          | sample:{<sample-name>}; attribute:{<path-to-attribute>}; reason:must match pattern {<pattern>}                                                                                             |
+| Missing mandatory field "{<property-name>}".                                                                                                                                     | sample:{<sample-name>}; attribute:{<path-to-attribute>}; reason:must have required property {<property-name>}                                                                              |
+| Missing recommended field "{<property-name>}".                                                                                                                                   | **DEPRECATED**. Recommended field validation will no longer be supported.                                                                                                                  |
```

Comment:
I don't understand. Did missing recommended fields appear as errors in the XML validation?

Reply:
In the file I got with the list of errors, it is listed (`ERAM.2.1.8=Missing recommended field "{0}".`)

Reply:
These read fine to me. What seems to be missing though is the prefix "ERROR: " or similar, then users clearly see that it is an error message.
 
( I had to read the read the markdown in my IDE, to check it was functional md and to read it properly, which it is)

Reply:
@Woolly-at-EBI the XML receipt that the users get after validation is usually wrapped in `<ERROR> message </ERROR>`. e.g.:

**Current message**
```xml
<ERROR>Missing mandatory field "geographic location (country and/or sea)".</ERROR>
```
**After changing validation**
```xml
<ERROR>sample:a803e2e8-6fcb-475e-ba2c-1ea3f4637527;attribute:geographic location (country and/or sea);reason:must have required property 'geographic location (country and/or sea)</ERROR>
```

Would that be enough, or do we want the error message to reflect that it's an error?